src/server/scripts/Events/brewfest.cpp

```diff
@@ -2057,6 +2064,52 @@ class spell_direbrew_disarm : public AuraScript
     }
 };
 
+struct npc_wild_wolpertinger : public ScriptedAI
+{
+public:
+    npc_wild_wolpertinger(Creature* creature) : ScriptedAI(creature) { }
+
+    ObjectGuid PlayerGUID;
+    Player* GetPlayer() { return ObjectAccessor::GetPlayer(*me, PlayerGUID); }
+
+    void Reset() override
+    {
+        me->RemoveUnitFlag(UNIT_FLAG_PACIFIED | UNIT_FLAG_DISABLE_MOVE);
+    }
+
+    void SpellHit(Unit* caster, SpellInfo const* spell) override
+    {
+        if (spell->Id == SPELL_WOLPERTINGER_NET)
+        {
+            PlayerGUID = caster->GetGUID();
+            me->SetReactState(REACT_PASSIVE);
+            me->CombatStop(false);
+            GetPlayer()->CombatStop();
```

Comment:
is player and caster not the same unit in this case?

Can't we do caster->ToPlayer()->CombatStop();?

Reply:
you're right, I wanted to make things easier